src/rgw/radosgw-admin/radosgw-admin.cc

```diff
@@ -1930,7 +1937,11 @@ static int send_to_url(const string& url,
   RGWRESTSimpleRequest req(g_ceph_context, info.method, url, NULL, &params, opt_region);
 
   bufferlist response;
-  int ret = req.forward_request(dpp(), key, info, MAX_REST_RESPONSE, &in_data, &response, null_yield);
+  auto result = req.forward_request(dpp(), key, info, MAX_REST_RESPONSE, &in_data, &response, null_yield);
+  if (!result) {
+    return result.error();
+  }
+  int ret = rgw_http_error_to_errno(*result);
 
   int parse_ret = parser.parse(response.c_str(), response.length());
   if (parse_ret < 0) {
```

Comment:
Should we still return `ret` in case parsing has failed? I'm thinking of a situation where an unexpected response has arrived, like from a loadbalancer like nginx (serving 5xx with HTML), and we don't return the actual error code but a parsing HTML error?

Reply:
oops, i had meant to preserve the existing error handling but did change `send_to_remote_gateway()` above. i agree and will change this one too